app/models/v_host.rb

```diff
@@ -101,15 +99,30 @@ def must_have_a_unencrypted_ssl_key
   end
   
   def must_have_a_well_formatted_ssl_key
-     OpenSSL::PKey::RSA.new(ssl_key) if errors[:ssl_key].empty?
+    tmp_cert = ssl_key.gsub("\n", "")
+    tmp_cert.strip =~ /^-+BEGIN EC PRIVATE KEY(.*)END EC PRIVATE KEY-+$/
+
+    if tmp_cert.strip =~ /^-+BEGIN EC PRIVATE KEY(.*)END EC PRIVATE KEY-+$/
+      OpenSSL::PKey::EC.new(ssl_key) if errors[:ssl_key].empty?
+    else
+      OpenSSL::PKey::RSA.new(ssl_key) if errors[:ssl_key].empty?
+    end
   rescue
     errors.add(:ssl_key, 'is invalid')
   end
   
   def private_key_must_match_ssl_certificate
     if errors.empty?
-      key = OpenSSL::PKey::RSA.new ssl_key
       cert = OpenSSL::X509::Certificate.new ssl_certificate
+      
+      tmp_cert = ssl_key.gsub("\n", "")
+      tmp_cert.strip =~ /^-+BEGIN EC PRIVATE KEY(.*)END EC PRIVATE KEY-+$/
+      
+      if tmp_cert.strip =~ /^-+BEGIN EC PRIVATE KEY(.*)END EC PRIVATE KEY-+$/
+        key = OpenSSL::PKey::EC.new(ssl_key) if errors[:ssl_key].empty?
+      else
+        key = OpenSSL::PKey::RSA.new(ssl_key) if errors[:ssl_key].empty?
+      end
```

Comment:
same here: https://ruby-doc.org/stdlib-2.5.1/libdoc/openssl/rdoc/OpenSSL/PKey.html

you can test this by using the test key elow like this:

```
OpenSSL::PKey.read PLAIN_EC_KEY
```